packages/compass-data-modeling/src/components/export-diagram-modal.tsx

```diff
@@ -64,25 +66,68 @@ const ExportDiagramModal = ({
   const [exportFormat, setExportFormat] = useState<'png' | 'json' | null>(null);
   const diagram = useDiagram();
   const [isExporting, setIsExporting] = useState(false);
+  const abortControllerRef = useRef<AbortController | null>(null);
+  const toast = useToast();
+  useEffect(() => {
+    const cleanup = () => {
+      if (abortControllerRef.current) {
+        abortControllerRef.current.abort();
+        abortControllerRef.current = null;
+      }
+    };
+    const abortController = new AbortController();
+    if (isModalOpen) {
+      abortControllerRef.current = abortController;
+    } else {
+      cleanup();
+    }
+    return cleanup;
+  }, [isModalOpen]);
+
+  const onClose = useCallback(() => {
+    setExportFormat(null);
+    setIsExporting(false);
+    abortControllerRef.current?.abort();
+    abortControllerRef.current = null;
+    onCloseClick();
+  }, [onCloseClick]);
 
   const onExport = useCallback(async () => {
-    if (!exportFormat || !model) {
-      return;
+    try {
+      if (!exportFormat || !model) {
```

Comment:
Would it make more sense to have these booleans disable the export button instead? That way a user doesn't click export and nothing happens.